contracts/finance/VestingWalletConfidentialFactory.sol

```diff
@@ -0,0 +1,213 @@
+        uint48 cliffSeconds,
+        address executor
+    );
+
+    struct VestingPlan {
+        address beneficiary;
+        externalEuint64 encryptedAmount;
+        uint48 start;
+        uint48 cliff;
+        address executor;
+    }
+
+    constructor() {
+        _vestingImplementation = address(new VestingWalletCliffExecutorConfidential());
+    }
+
+    /**
+     * @dev Batches the funding of multiple confidential vesting wallets.
+     *
+     * Funds are sent to predeterministic wallet addresses. Wallets can be created either
+     * before or after this operation.
+     *
+     * Emits a single {VestingWalletConfidentialBatchFunded} event in addition to multiple
+     * {VestingWalletConfidentialFunded} events related to funded vesting plans.
+     */
+    function batchFundVestingWalletConfidential(
+        address confidentialFungibleToken,
+        VestingPlan[] calldata vestingPlans,
+        uint48 durationSeconds,
+        bytes calldata inputProof
+    ) public virtual returns (euint64 totalTransferedAmount) {
+        require(durationSeconds > 0, VestingWalletConfidentialInvalidDuration());
+        totalTransferedAmount = euint64.wrap(0);
+        for (uint256 i = 0; i < vestingPlans.length; i++) {
+            VestingPlan memory vestingPlan = vestingPlans[i];
+            euint64 encryptedAmount = FHE.fromExternal(vestingPlan.encryptedAmount, inputProof);
+            require(
+                vestingPlan.start >= block.timestamp,
+                VestingWalletConfidentialInvalidStartTimestamp(vestingPlan.beneficiary, vestingPlan.start)
+            );
+            address vestingWalletConfidential = predictVestingWalletConfidential(
+                vestingPlan.beneficiary,
+                vestingPlan.start,
+                durationSeconds,
+                vestingPlan.cliff,
+                vestingPlan.executor
+            );
+            FHE.allowTransient(encryptedAmount, confidentialFungibleToken);
+            euint64 transferredAmount = IConfidentialFungibleToken(confidentialFungibleToken).confidentialTransferFrom(
+                msg.sender,
+                vestingWalletConfidential,
+                encryptedAmount
+            );
+            totalTransferedAmount = FHE.select(
+                FHE.eq(encryptedAmount, transferredAmount),
+                FHE.add(totalTransferedAmount, transferredAmount),
+                FHE.asEuint64(0)
+            );
+            emit VestingWalletConfidentialFunded(
+                vestingWalletConfidential,
+                vestingPlan.beneficiary,
+                confidentialFungibleToken,
+                transferredAmount,
+                vestingPlan.start,
+                durationSeconds,
+                vestingPlan.cliff,
+                vestingPlan.executor
+            );
+        }
+        FHE.allow(totalTransferedAmount, msg.sender);
+        emit VestingWalletConfidentialBatchFunded(msg.sender, totalTransferedAmount);
```

Comment:
Given that this can be determined offchain (by adding all the transferAmounts) and that it is quite expensive to get, I'd forgo this calculation.